cmd/juju/cloud/updatecloud_test.go

```diff
@@ -0,0 +1,161 @@
+	c.Assert(err, jc.ErrorIsNil)
+}
+
+func (s *updateCloudSuite) TestUpdateLocalCacheFromFile(c *gc.C) {
+	cmd, fileName := s.setupCloudFileScenario(c, func(controllerName string) (cloud.UpdateCloudAPI, error) {
+		return nil, errors.New("")
+	})
+	_, err := cmdtesting.RunCommand(c, cmd, "garage-maas", "-f", fileName)
+	c.Assert(err, jc.ErrorIsNil)
+	c.Assert(s.api.Calls(), gc.HasLen, 0)
+}
+
+func (s *updateCloudSuite) TestUpdateLocalCacheBadFile(c *gc.C) {
+	fake := newFakeCloudMetadataStore()
+	badFileErr := errors.New("")
+	fake.Call("ParseCloudMetadataFile", "somefile.yaml").Returns(map[string]jujucloud.Cloud{}, badFileErr)
+
+	addCmd := cloud.NewUpdateCloudCommandForTest(fake, s.store, nil)
+	_, err := cmdtesting.RunCommand(c, addCmd, "cloud", "-f", "somefile.yaml")
+	c.Check(errors.Cause(err), gc.Equals, badFileErr)
+}
+
+func (s *updateCloudSuite) TestUpdateControllerFromFile(c *gc.C) {
+	var controllerNameCalled string
+	cmd, fileName := s.setupCloudFileScenario(c, func(controllerName string) (cloud.UpdateCloudAPI, error) {
+		controllerNameCalled = controllerName
+		return s.api, nil
+	})
+	_, err := cmdtesting.RunCommand(c, cmd, "garage-maas", "-f", fileName, "--controller", "mycontroller")
+	c.Assert(err, jc.ErrorIsNil)
+	s.api.CheckCallNames(c, "UpdateCloud", "Close")
+	c.Assert(controllerNameCalled, gc.Equals, "mycontroller")
+	s.api.CheckCall(c, 0, "UpdateCloud", jujucloud.Cloud{
+		Name:        "garage-maas",
+		Type:        "maas",
+		Description: "Metal As A Service",
+		AuthTypes:   jujucloud.AuthTypes{"oauth1"},
+		Endpoint:    "http://garagemaas",
+	})
+}
+
+func (s *updateCloudSuite) TestUpdateControllerLocalCacheBadFile(c *gc.C) {
+	fake := newFakeCloudMetadataStore()
+	badFileErr := errors.New("")
+	fake.Call("ParseCloudMetadataFile", "somefile.yaml").Returns(map[string]jujucloud.Cloud{}, badFileErr)
+
+	addCmd := cloud.NewUpdateCloudCommandForTest(fake, s.store, nil)
+	_, err := cmdtesting.RunCommand(c, addCmd, "cloud", "-f", "somefile.yaml", "--controller", "mycontroller")
+	c.Check(errors.Cause(err), gc.Equals, badFileErr)
+}
+
+func (s *updateCloudSuite) TestUpdateControllerFromLocalCache(c *gc.C) {
+	s.createLocalCacheFile(c)
+	var controllerNameCalled string
+	cmd, _ := s.setupCloudFileScenario(c, func(controllerName string) (cloud.UpdateCloudAPI, error) {
+		controllerNameCalled = controllerName
+		return s.api, nil
+	})
+	_, err := cmdtesting.RunCommand(c, cmd, "garage-maas", "--controller", "mycontroller")
+	c.Assert(err, jc.ErrorIsNil)
+	s.api.CheckCallNames(c, "UpdateCloud", "Close")
+	c.Assert(controllerNameCalled, gc.Equals, "mycontroller")
+	s.api.CheckCall(c, 0, "UpdateCloud", jujucloud.Cloud{
+		Name:        "garage-maas",
+		Type:        "maas",
+		Description: "Metal As A Service",
+		AuthTypes:   jujucloud.AuthTypes{"oauth1"},
+		Endpoint:    "http://garagemaas",
+	})
+}
+
+type fakeUpdateCloudAPI struct {
+	jujutesting.Stub
+	updatedClouds []jujucloud.Cloud
```

Comment:
We set this but never check it.
We either need to record what was done and check that or use the stub to check the calls

Reply:
Good point, the raiseError was unused too (and we have NextErr() at any rate.). Calls are checked in the tests.